app/lib/cms/system_name.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module CMS::SystemName
+
+  extend ActiveSupport::Concern
+
+  included do
+    validates :system_name, uniqueness: { scope: %i[provider_id type], allow_blank: true },
+              format: { with: %r{\A\w[\w\-/_]+\z}, allow_blank: true }, length: { maximum: 255 },
+              presence: true
+
+    before_validation :set_system_name, on: :create
+  end
+
+  protected
+
+  def set_system_name
+    self.system_name = title&.parameterize if title && system_name.blank?
```

Comment:
Changed it to

`self.system_name ||= title.parameterize if title`

 if title can't be removed, otherwise in case of not having title will `system_name` no `nil`

Reply:
Due to a test failing, I had to revert this to:

`self.system_name = title.parameterize if title.present? && system_name.blank?`

`||=` doesn't work properly when `system_name = ""`, so `.blank?` is required